packages/ui-components/src/lib/inputs/InputSelect/InputSelect.tsx

```diff
@@ -120,19 +155,29 @@ export const InputSelect: FC<InputSelectProps> = ({
     setValue(controlledValue);
   }, [controlledValue]);
 
+  const onAnimationStart = ({ animationName }: AnimationEvent) => {
+    if (animationName === 'mui-auto-fill') return setAutoFilling(true);
+    if (animationName === 'mui-auto-fill-cancel') return setAutoFilling(false);
+  };
+
+  // Note: We need to manually handle the browser autofill event by listening for 'animationstart' and handling it based on the
+  // animation name. This is because there is currently no native way to listen explicitly for autofill events.
```

Comment:
@jaredhill4 are there helpful documentation or stack overflows for this. Kind mind blowing.

Reply:
@jaruesink,

Yea, there's a few articles:
* https://codedaily.io/tutorials/Animated-Input-Label-with-Chrome-Autofill-Detection-in-React
* https://medium.com/@brunn/detecting-autofilled-fields-in-javascript-aed598d25da7

Reply:
@jaruesink,

I've made a few more updates and cleaned things up a bit.